src/Test/ApiTestCase.php

```diff
@@ -0,0 +1,252 @@
+<?php
+
+namespace Lakion\ApiTestCase\Test;
+
+use Coduo\PHPMatcher\Factory\SimpleFactory;
+use Psr\Http\Message\ResponseInterface;
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+use Symfony\Component\Finder\Finder;
+
+/**
+ * @author Łukasz Chruściel <lukasz.chrusciel@lakion.com>
+ * @author Paweł Jędrzęjewski <pawel.jedrzejewski@lakion.com>
+ * @author Michał Marcinkowski <michal.marcinkowski@lakion.com>
+ */
+abstract class ApiTestCase extends WebTestCase
+{
+    /**
+     * @var \Symfony\Bundle\FrameworkBundle\Client $client
+     */
+    protected $client;
+
+
+    public function setUp()
+    {
+        $this->client = static::createClient(array(), array('HTTP_Accept' => 'application/json'));
+    }
+
+    public function tearDown()
+    {
+        if (null !== $this->client && null !== $this->client->getContainer()) {
+            foreach ($this->client->getContainer()->getMockedServices() as $id => $service) {
+                $this->client->getContainer()->unmock($id);
+            }
+        }
+
+        \Mockery::close();
+        $this->client = null;
+
+        parent::tearDown();
+    }
+
+    /**
+     * Asserts that response has JSON content.
+     *
+     * @param ResponseInterface $response
+     */
+    protected function assertJsonResponse(ResponseInterface $response)
+    {
+        $this->assertTrue(
+            in_array(
+                'application/json',
+                $response->getHeader('Content-Type')
+            )
+        );
+    }
+
+    /**
+     * Asserts that response has given status code.
+     *
+     * @param ResponseInterface $response
+     * @param integer $statusCode
+     */
+    protected function assertResponseCode(ResponseInterface $response, $statusCode = 200)
+    {
+        $this->assertEquals($statusCode, $response->getStatusCode());
+    }
+
+    /**
+     * Asserts that response has JSON content matching the one given in file.
+     *
+     * @param ResponseInterface $response
+     * @param string $filename
+     */
+    protected function assertJsonResponseContent(ResponseInterface $response, $filename)
+    {
+        $expectedResponse = file_get_contents(__DIR__.sprintf('/../Tests/Responses/%s.json', $filename));
+        $actualResponse = $response->getContent();
+
+        $actualResponse = json_encode(json_decode($actualResponse), JSON_PRETTY_PRINT);
+
+        $factory = new SimpleFactory();
+        $matcher = $factory->createMatcher();
+
+        $result = $matcher->match($actualResponse, $expectedResponse);
+
+        if (!$result) {
+            echo $matcher->getError();
+
+            $expectedResponse = explode(PHP_EOL, (string) $expectedResponse);
+            $actualResponse   = explode(PHP_EOL, (string) $actualResponse);
+
+            $diff = new \Diff($expectedResponse, $actualResponse, array());
+
+            $renderer = new \Diff_Renderer_Text_Unified;
+            echo $diff->render($renderer);
```

Comment:
If we throw exception (see my comment below), then we don't need to do  this ugly echo.